### PR TITLE
Recalculate bbox when LoD filter is used for CityGML exports

### PR DESCRIPTION
**Pull request by clausnagel**
This PR addresses #104.

### CODE REVIEW

File: impexp-core/src/main/java/org/citydb/core/operation/exporter/concurrent/DBExportWorker.java [hunk 1]
```diff
@@ -184,6 +190,11 @@ private void doWork(DBSplittingResult work) {
 					AbstractGML object = exporter.exportObject(work.getId(), work.getObjectType());
 					if (object instanceof AbstractFeature) {
 						feature = (AbstractFeature) object;
+
+						if (useLodFilter && feature.isSetBoundedBy()) {
+							feature.setBoundedBy(feature.calcBoundedBy(bboxOptions));
```
Comment (yaozhihang) [inline impexp-core/src/main/java/org/citydb/core/operation/exporter/concurrent/DBExportWorker.java:195]: The feature here is top-level. How about sub-features, should their bbox also be recalculated?

Comment (clausnagel) [reply impexp-core/src/main/java/org/citydb/core/operation/exporter/concurrent/DBExportWorker.java:198]: e8097757143c409f6f67d3a2da7650d71fd52644

**Comment by yaozhihang** [comment]
There is an option `generalOptions.getEnvelope().isUseEnvelopeOnCityModel()`. Maybe it should also be taken into account.

**Comment by clausnagel** [comment]
> There is an option `generalOptions.getEnvelope().isUseEnvelopeOnCityModel()`. Maybe it should also be taken into account.

Well, this option controls whether a `gml:Envelope` shall be created for the root `<CityModel>` in the export dataset. Since `<CityModel>` this is the first line written to the dataset, we cannot compute its envelope based on the exported city objects but only on the envelopes stored in the database.

Or did you mean something else?

**Comment by yaozhihang** [comment]
Yes, that is what I mean. OK, it sounds impossible to reccalculate the CityModel envelope based on the LOD filter.

**Review by yaozhihang — APPROVED** [review]

**Comment by clausnagel** [comment]
Well, not impossible in general. But this would require changing the dataset after it has been written. This cannot be done if we write to a stream though. In such cases, we would need some temporary export datasets which is only sent to the stream once it has been adapted. Thus, the stream would have to wait long for a response... So no easy and nice solution, at least.

**Comment by yaozhihang** [comment]
yes, we could improve this in future in a separate PR once we find a better solution.

**Pull request merged**